specifyweb/frontend/js_src/lib/components/FormSliders/AttachmentsCollection.tsx

```diff
@@ -44,9 +44,19 @@ export function AttachmentsCollection({
     })
   );
 
+  const [isAttachmmentsSaved, setIsAttachmmentsSaved] = React.useState(false);
+
+  React.useEffect(() => {
+    const savedAttachments = attachments.filter(
+      (attachment) => attachment.attachmentLocation !== null
+    );
+    setIsAttachmmentsSaved(savedAttachments.length === attachments.length);
```

Comment:
this can be simplified if you use https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some

take some time to look at all the array methods there are and learn to use them - arrays are very common in JavaScript, so this will make your life easier
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array#array_methods_and_empty_slots